leo/cli/commands/common/util.rs

```diff
@@ -0,0 +1,116 @@
+// Copyright (C) 2019-2025 Provable Inc.
+// This file is part of the Leo library.
+
+// The Leo library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The Leo library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with the Leo library. If not, see <https://www.gnu.org/licenses/>.
+
+use leo_package::{Package, ProgramData};
+use leo_span::Symbol;
+
+use indexmap::IndexSet;
+use std::path::PathBuf;
+use walkdir::WalkDir;
+
+/// Collects paths to Leo source files for each program in the package.
+///
+/// For each non-test program, it searches the `src` directory for `.leo` files.
+/// It separates the `main.leo` file from the rest and returns a tuple:
+/// (`main.leo` path, list of other `.leo` file paths).
+/// Test programs are included with an empty list of additional files.
+/// Programs with bytecode data are ignored.
+///
+/// # Arguments
+/// * `package` - Reference to the package containing programs.
+///
+/// # Returns
+/// A vector of tuples with the main file and other source files.
+pub fn collect_leo_paths(package: &Package) -> Vec<(PathBuf, Vec<PathBuf>)> {
```

Comment:
Do you see an opportunity to also use this in `build.rs`? Not needed for this PR.